ckanext/opendatani/controller.py

```diff
@@ -249,3 +254,57 @@ def resource_read(self, id, resource_id):
                     h.flash_error('Sorry, this file is too large to be able to display in the browser, please download the data resource to examine it further.')
         template = self._resource_template(dataset_type)
         return render(template, extra_vars=vars)
+
+
+class CustomReportController(CorePackageController):
+    def prepare_report(self, org):
+        """
+        Creates a CSV publisher report and returns it as a string
+        :param org: organization
+        :type org: string
+        :return: a CSV string
+        :rtype: string
+        """
+
+        try:
+            data_dict = {'org_name': org}
+
+            if org == 'full-portal':
+                data_dict = {}
+
+            resource = toolkit.get_action(
+                'report_resources_by_organization')({}, data_dict)
+
+            # We need this in case no datasets exist for the given organization
+            # or the organization doesn't exist
+            if not resource:
+                toolkit.abort(404, _('Either {0} does not exist, or it \
+                                     has no datasets.'.format(org)))
+
+            csvout = StringIO.StringIO()
+            csvwriter = csv.writer(
+                csvout,
+                dialect='excel',
+                quoting=csv.QUOTE_NONNUMERIC
+            )
+
+            fields = resource[0].keys()
```

Comment:
What if `resource` is empty? Can you add a check for that?

Reply:
`if not resource` takes care of this. Empty will return false.